include/inexor/vulkan-renderer/world/collision_query.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/world/collision.hpp"
+#include "inexor/vulkan-renderer/world/cube.hpp"
+
+#include <glm/vec3.hpp>
+
+#include <optional>
+
+namespace inexor::vulkan_renderer::world {
+
+// TODO: Implement PointCubeCollision
+class OctreeCollisionQuery {
+    const Cube &m_cube;
+
+    /// @brief ``True`` if the ray collides with the cube's bounding sphere.
+    /// @param pos The start position of the ray.
+    /// @param dir The direction of the ray.
+    /// @return ``True`` if the ray collides with the octree cube's bounding sphere.
+    [[nodiscard]] bool ray_sphere_collision(glm::vec3 pos, glm::vec3 dir) const;
+
+    /// @brief ``True`` of the ray build from the two vectors collides with the cube's bounding box.
+    /// @note There is no such function as glm::intersectRayBox.
+    /// @param box_bounds An array of two vectors which represent the edges of the bounding box.
+    /// @param pos The start position of the ray.
+    /// @param dir The direction of the ray.
+    /// @return ``True`` if the ray collides with the octree cube's bounding box.
+    [[nodiscard]] bool ray_box_collision(std::array<glm::vec3, 2> box_bounds, glm::vec3 pos, glm::vec3 dir) const;
+
+public:
+    /// @brief Create a new octree collision query
+    /// @param world The octree world to check in this collision.
+    OctreeCollisionQuery(const Cube &world);
```

Comment:
Should the constructor be placed on top?

Reply:
Yes. Actually the entire code needs proper sorting of declarations: https://github.com/inexorgame/vulkan-renderer/issues/359